client/gutenberg/extensions/simple-payments/edit.js

```diff
@@ -113,7 +113,7 @@ class SimplePaymentsEdit extends Component {
 		this.setState( { isSavingProduct: true }, async () => {
 			saveEntityRecord( 'postType', SIMPLE_PAYMENTS_PRODUCT_POST_TYPE, this.toApi() )
 				.then( record => {
-					setAttributes( { paymentId: record.id } );
+					record && setAttributes( { paymentId: record.id } );
```

Comment:
I'd prefer a "real" conditional branch here:
```suggestion
					if ( record ) {
						setAttributes( { paymentId: record.id } );
					}
					return record;
```
It's also nice to return whatever we get in the promise so we could continue to chain.